rpc/blockchain.go

```diff
@@ -21,23 +23,43 @@ import (
 	v2 "github.com/harmony-one/harmony/rpc/v2"
 	"github.com/harmony-one/harmony/shard"
 	stakingReward "github.com/harmony-one/harmony/staking/reward"
+
+	lru "github.com/hashicorp/golang-lru"
 )
 
 // PublicBlockchainService provides an API to access the Harmony blockchain.
 // It offers only methods that operate on public data that is freely available to anyone.
 type PublicBlockchainService struct {
-	hmy     *hmy.Harmony
-	version Version
+	hmy        *hmy.Harmony
+	version    Version
+	limiter    *rate.Limiter
+	blockCache *lru.Cache
 }
 
+const (
+	DefaultRateLimiterWaitTimeout = 5 * time.Second
+	blockCacheLimit               = 256
+)
+
 // NewPublicBlockchainAPI creates a new API for the RPC interface
-func NewPublicBlockchainAPI(hmy *hmy.Harmony, version Version) rpc.API {
-	return rpc.API{
-		Namespace: version.Namespace(),
-		Version:   APIVersion,
-		Service:   &PublicBlockchainService{hmy, version},
-		Public:    true,
+func NewPublicBlockchainAPI(hmy *hmy.Harmony, version Version, limiterEnable bool, limit int) rpc.API {
+	blockCache, _ := lru.New(blockCacheLimit)
+	if limiterEnable {
```

Comment:
What is the case for not enabling rate limiter? Is there going to be a transition period or something?

Reply:
I replied on rate limiter PR

Reply:
@JackyWYX  Now for the explorer node for the time being there is no unrestricted scenario, but I think this switch is still exposed as much as possible, this is a functional switch

Reply:
OK, sure. But do we really need to expose this to CLI level? @LeoHChen